src/dsp_tools/commands/xmlupload/check_consistency_with_ontology.py

```diff
@@ -149,3 +164,66 @@ def _get_separate_prefix_and_iri_from_onto_prop_or_cls(
         return tuple(prop_or_cls.split(":"))
     else:
         return None, None
+
+
+def _analyse_if_all_text_value_encodings_are_correct(
+    root: etree._Element, text_prop_look_up: TextValuePropertyGUI
+) -> None:
+    text_value_in_data = _get_all_ids_prop_encoding_from_root(root)
+    all_checked = [x for x in text_value_in_data if not _check_correctness_one_prop(x, text_prop_look_up)]
+    if len(all_checked) == 0:
+        return None
+    msg, df = InvalidTextValueEncodings(all_checked).execute_problem_protocol()
+    if df is not None:
+        csv_file = f"text_value_encoding_errors_{datetime.now().strftime('%Y-%m-%d_%H%M%S')}.csv"
+        df.to_csv(path_or_buf=Path(Path.cwd(), csv_file), index=False)
+        msg += (
+            "\n\n---------------------------------------\n\n"
+            f"\nAll the problems are listed in the file: '{Path.cwd()}/{csv_file}'"
+        )
+    raise UserError(msg)
+
+
+def _get_all_ids_prop_encoding_from_root(root: etree._Element) -> list[TextValueData]:
+    res_list = []
+    for res_input in root.iterchildren(tag="resource"):
+        if (res_result := _get_id_prop_encoding_from_one_resource(res_input)) is not None:
+            res_list.extend(res_result)
+    return res_list
+
+
+def _get_id_prop_encoding_from_one_resource(resource: etree._Element) -> list[TextValueData] | None:
+    if not (children := list(resource.iterchildren(tag="text-prop"))):
+        return None
+    res_id = resource.attrib["id"]
+    return [_get_prop_encoding_from_one_property(res_id, child) for child in children]
+
+
+def _get_prop_encoding_from_one_property(res_id: str, property: etree._Element) -> TextValueData:
+    prop_name = property.attrib["name"]
+    child_attrib = [x.attrib for x in property.iterchildren()]
+    encodings = {x.get("encoding") for x in child_attrib}
+    return TextValueData(res_id, prop_name, encodings)
+
+
+def _check_correctness_one_prop(text_val: TextValueData, text_prop_look_up: TextValuePropertyGUI) -> bool:
+    text_encoding = text_val.encoding
```

Comment:
I am not a fan of either of the two options (one of them commented out).
Ruff is happy with the first one, but does not accept the second.
While the second is ugly I wonder if it is easier to understand?
What would your preference be?

Reply:
Is the problem perhaps the underlying data structure you use? Does it make sense that the set can contain `None` AND `utf-8` at the same time? 

So if you go to the code where this set is constructed, perhaps you find a way to write only strings into this set.

Reply:
Regarding mixed types:
Yes it is possible, it really should not be but in our test data we had a case of (wrong) mixed types: https://github.com/dasch-swiss/dsp-tools/blob/443195f68023eb267e34568e0fa9a215886d3944/testdata/xml-data/test-data-systematic.xml#L186

Regarding `None`:
I misremembered this, I knew we don't have to specify an encoding and then it is by default utf8, I though this default value would happen during the xmlupload and not the excel2xml.make_text_prop. My bad, I will fix that and remove it as an option. 

Regarding an Enum class:
I have thought about that but it seemed overkill for so little options. I look into doing it with a union type.

Reply:
Regarding mixed types:  
if it is no longer in the test data, could the xml schema be adjusted to enforce that it never happens?

Enum/Union:  
Yes... python enums are not great, too. But if there are only two possible states, you can simply capture the information with a boolean. That's the easiest, I suppose

Reply:
I will look into fixing the xml schema now, and would do that first, so I can then simplify the logic here. Thanks for your input!